src/main/java/com/employeemanagement/employeemanagement/service/EmployeeService.java

```diff
@@ -81,20 +81,93 @@ public void create(EmployeeDTO employeeDTO) {
-		String responseMessage = "Collaborator of ID " + employeeDTO.getId() + " not found";
-
-		if (defaultEmployee != null) {
-			defaultEmployee.setFirstName(employeeDTO.getFirstName());
-			defaultEmployee.setMiddleName(employeeDTO.getMiddleName());
-			defaultEmployee.setLastName(employeeDTO.getLastName());
-			defaultEmployee.setCpf(employeeDTO.getCpf());
-			defaultEmployee.setCategory(employeeDTO.getCategory());
-			create(EmployeeDTO.convertToDTO(defaultEmployee));
-			responseMessage = "Employee of ID " + employeeDTO.getId() + " updated successfully!";
-			return responseMessage;
+		Employee defaultEmployee = getEmployeeRepository().findById(employeeDTO.getId())
+				.orElseThrow(() -> new RuntimeException("Collaborator of ID " + employeeDTO.getId() + " not found"));
+
+		defaultEmployee.setFirstName(
+				employeeDTO.getFirstName() != null && !employeeDTO.getFirstName().isEmpty() ? employeeDTO.getFirstName()
+						: defaultEmployee.getFirstName());
+		defaultEmployee.setMiddleName(employeeDTO.getMiddleName() != null && !employeeDTO.getMiddleName().isEmpty()
+				? employeeDTO.getMiddleName()
+				: defaultEmployee.getMiddleName());
+		defaultEmployee.setLastName(
+				employeeDTO.getLastName() != null && !employeeDTO.getLastName().isEmpty() ? employeeDTO.getLastName()
+						: defaultEmployee.getLastName());
+		defaultEmployee.setCpf(employeeDTO.getCpf() != null && !employeeDTO.getCpf().isEmpty() ? employeeDTO.getCpf()
+				: defaultEmployee.getCpf());
+		defaultEmployee.setCategory(
+				employeeDTO.getCategory() != null && !employeeDTO.getCategory().isEmpty() ? employeeDTO.getCategory()
+						: defaultEmployee.getCategory());
+
+		getEmployeeRepository().save(defaultEmployee);
+
+		if (employeeDTO.getTrainingsId() != null) {
+			for (Long trainingId : employeeDTO.getTrainingsId()) {
+				Training training = getTrainingRepository().findById(trainingId)
+						.orElseThrow(() -> new RuntimeException("Training of ID " + trainingId + " not found"));
+
+				EmployeeTraining relationship = new EmployeeTraining();
+				relationship.setTraining(training);
+				Status status = getStatusRepository().findById(1L)
+						.orElseThrow(() -> new RuntimeException("Status of ID 1 not found"));
+				relationship.setStatus(status);
+				relationship.setEmployee(defaultEmployee);
+
+				employeeTrainingRepository.save(relationship);
+			}
 		}
-		return responseMessage;
+
+		return "Employee of ID " + employeeDTO.getId() + " updated successfully!";
+	}
+
+	public String inProgress(Long iDEmployee, Long iDTraining) {
+	    Employee defaultEmployee = getEmployeeRepository().findById(iDEmployee).orElse(null);
+	    if (defaultEmployee != null) {
+	        for (EmployeeTraining training : defaultEmployee.getTrainings()) {
+	            if (training.getTraining().getId() == iDTraining) {
+	                if (training.getStatus().getId() < 2L) {
+	                    employeeTrainingRepository.delete(training);
```

Comment:
Redundância de IFs, "complexidade de IFs", da pra melhorar  utilizando apenas um IF.
Dica: você pode criar um método customizado no repositório de EmployeeTraining para buscar por Id de employee e training sem precisar usar o For.